jobs/build/aws-ami/Jenkinsfile

```diff
@@ -0,0 +1,117 @@
+        env.WORKSPACE = pwd()
+    }
+}
+
+// Expose properties for a parameterized build
+properties(
+        [
+                disableConcurrentBuilds(),
+                [$class: 'ParametersDefinitionProperty',
+                 parameterDefinitions:
+                         [
+                                 [$class: 'hudson.model.StringParameterDefinition', defaultValue: 'kwoodson@redhat.com', description: 'Success Mailing List', name: 'MAIL_LIST_SUCCESS'],
+                                 [$class: 'hudson.model.StringParameterDefinition', defaultValue: 'kwoodson@redhat.com', description: 'Failure Mailing List', name: 'MAIL_LIST_FAILURE'],
+                                 [$class: 'hudson.model.BooleanParameterDefinition', defaultValue: false, description: 'Force rebuild even if no changes are detected?', name: 'FORCE_REBUILD'],
+                                 [$class: 'hudson.model.ChoiceParameterDefinition',
+                                    choices: "openshift-enterprise\norigin",
+                                    defaultValue: 'openshift-enterprise',
+                                    description: '''origin                    Openshift origin  <br>
+                                                    openshift-enterprise      Openshift Enterpriseonline <br>
+                                    ''',
+                                    name: 'DEPLOYMENT_TYPE'],
+                                 [$class: 'hudson.model.StringParameterDefinition', defaultValue: 'opstest', description: 'Clusterid to set for ami build. Also used for VPC and other settings', name: 'CLUSTERID'],
+                                 [$class: 'hudson.model.StringParameterDefinition', defaultValue: '3.7.0', description: 'Release version (matches version in branch name for release builds)', name: 'RELEASE_VERSION'],
+                                 [$class: 'BooleanParameterDefinition', defaultValue: false, description: 'Mock run to pickup new Jenkins parameters?.', name: 'MOCK'],
+                                 [$class: 'hudson.model.StringParameterDefinition', defaultValue: 'ami-068a787c', description: 'Base AMI id to build from.', name: 'BASE_AMI_ID'],
+                                 [$class: 'hudson.model.StringParameterDefinition', defaultValue: 'ami_base', description: 'Base AMI instance name.', name: 'BASE_AMI_NAME'],
+                                 [$class: 'hudson.model.StringParameterDefinition', defaultValue: 'ami_builder_key', description: 'Name of the AWS SSH key user to use.', name: 'AWS_SSH_KEY_USER'],
+                                 [$class: 'hudson.model.StringParameterDefinition', defaultValue: 'https://mirror.openshift.com/enterprise/online-int/latest/x86_64/os/', description: 'Base url for repository.', name: 'YUM_BASE_URL'],
+                         ]
+                ],
+        ]
+)
+
+// Force Jenkins to fail early if this is the first time this job has been run/and or new parameters have not been discovered.
```

Comment:
You can use this pattern, but I'm replacing it with MOCK in all our pipelines. The old pattern forced an error when a new property was introduced by trying to print its value early in the job. Now, I just run once with MOCK=true to pick up the new variables.

Reply:
Sounds good.  This was lifted from a separate JenkinsFile.